File: sol/solution/8 Kyu/55f1b763dd670651620000ce/Solution.py
#
# Solution
#
def count_char_occurrences(strng, char):
    return strng.count(char)





#
# Sample Tests
#
@test.describe("count test")
def tests():
    @test.it("basic tests")
    def basic():
        test.assert_equals(count_char_occurrences("missippi", 'i'), 3)
        test.assert_equals(count_char_occurrences("feed", 'e'), 2)
        test.assert_equals(count_char_occurrences("aaaaaaaa", 'a'), 8)
        test.assert_equals(count_char_occurrences("quicksilver", 'z'), 0)

File: sol/solution/8 Kyu/55f1b763dd670651620000ce/Solution.java
/*
Solution
*/
interface Count {
    static int countCharOccurrences(String s, char c) {
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c) {
                count++;
            }
        }
        return count;
    }
}





/*
Sample Tests
*/
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class CountTest {
  @Test
  void test_some_characters() {
    assertEquals(3, Count.countCharOccurrences("missippi", 'i'));
    assertEquals(2, Count.countCharOccurrences("feed", 'e'));
  }

  @Test
  void test_all_characters() {
    assertEquals(8, Count.countCharOccurrences("aaaaaaaa", 'a'));
  }

  @Test
  void test_no_characters() {
    assertEquals(0, Count.countCharOccurrences("quicksilver", 'z'));
  }
}